### PR TITLE
cupyx/scipy.signal: port *ord filter design functions from scipy.signal

### PR DESCRIPTION
**Pull request by ev-br**
cross-ref gh-7403

Port order selection functions for several types of filters
- [x] ellipord
- [x] cheb1ord
- [x] cheb2ord
- [x] buttord
.
- [x] band_stop_obj
- [x] optimize.fminbound replacement : a private `cupyx.scipy.signal._optimize` module
- [x] iirdesign

<s>This PR is on top of gh-7537</s>

### CODE REVIEW

File: tests/cupyx_tests/scipy_tests/signal_tests/test_filter_design.py [hunk 1]
```diff
@@ -0,0 +1,635 @@
...
+        return w, h
+
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_sosfreqz_design_ellip_3(self, xp, scp):
+        N, Wn = scp.signal.ellipord(0.3, 0.2, .5, 150)
+        sos = scp.signal.ellip(N, .5, 150, Wn, 'high', output='sos')
+        w, h = scp.signal.sosfreqz(sos)
+        return w, h
+
+    @testing.with_requires("mpmath > 0.10")
```
Comment (takagi) [inline tests/cupyx_tests/scipy_tests/signal_tests/test_filter_design.py:537]: I'll make another PR to make CI install mpmath.

**Comment by ev-br** [comment]
Rebased on main, now that gh-7537 is in.

**Review by takagi — APPROVED** [review]
LGTM!

**Comment by takagi** [comment]
I'll kick the CI after #7653 is merged.

File: tests/cupyx_tests/scipy_tests/signal_tests/test_iir_filter_design.py [hunk 2]
```diff
@@ -493,6 +494,478 @@ def test_ba_output(self, xp, scp):
...
+    def test_lowpass(self, xp, scp):
+        wp = 0.2
+        ws = 0.3
+        rp = 3
+        rs = 60
+        N, Wn = scp.signal.ellipord(wp, ws, rp, rs, False)
+        return N, Wn
+
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_lowpass_1000dB(self, xp, scp):
```
Comment (takagi) [inline tests/cupyx_tests/scipy_tests/signal_tests/test_iir_filter_design.py:774]:
```suggestion
    @testing.with_requires('scipy>=1.7')
    def test_lowpass_1000dB(self, xp, scp):
```

**Review by takagi — APPROVED** [review]

**Comment by takagi** [comment]
CI failures are not related.

**Comment by takagi** [comment]
Thanks, @ev-br!

**Pull request merged**